src/NUnitFramework/framework/Constraints/GreaterThanConstraint.cs

```diff
@@ -21,22 +21,37 @@
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 // ***********************************************************************
 
-using System;
-
 namespace NUnit.Framework.Constraints
 {
     /// <summary>
     /// Tests whether a value is greater than the value supplied to its constructor
     /// </summary>
     public class GreaterThanConstraint : ComparisonConstraint
     {
+        /// <summary>
+        /// The Description of what this constraint tests, for
+        /// use in messages and in the ConstraintResult.
+        /// </summary>
+        private string _description;
+        
         /// <summary>
         /// Initializes a new instance of the <see cref="GreaterThanConstraint"/> class.
         /// </summary>
         /// <param name="expected">The expected value.</param>
-        public GreaterThanConstraint(object expected) : base(expected)
+        public GreaterThanConstraint(object expected) : base(expected) {}
+
+        /// <summary>
+        /// The Description of what this constraint tests, for
+        /// use in messages and in the ConstraintResult.
+        /// </summary>
+        public override string Description
         {
-            Description = "greater than " + MsgUtils.FormatValue(expected);
+            get
+            {
+                if (System.String.IsNullOrEmpty(_description))
```

Comment:
Please use the `string` keyword rather than the `String` class name. But before you do that, why isn't this `_description == null`?
(Same in other files.)

Reply:
It seemed more "clear" to me to use `string.IsNullOrEmpty` rather than `_desciption == null`, which is probably better from the point of performance.

Reply:
If you use IsNullOrEmpty, it signals to the reader that there is a reason that both states (null and empty) might show up. I think we should decide that only one of those states is treated as 'uninitialized' so that there's less to keep in your head as you read through the code.

Reply:
Thank you for your advices. I appreciate it. Will try to follow them too in my next contributions.